cli/admin/src/main/java/io/pravega/cli/admin/dataRecovery/DurableLogInspectCommand.java

```diff
@@ -0,0 +1,369 @@
+import lombok.Cleanup;
+import lombok.Getter;
+import lombok.Setter;
+import lombok.val;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.BiConsumer;
+import java.util.function.Predicate;
+
+import static io.pravega.cli.admin.utils.FileHelper.createFileAndDirectory;
+
+/**
+ * This command provides an administrator with the basic primitives to inspect a DurableLog.
+ * The workflow of this command is as follows:
+ * 1. Reads the original DurableLog
+ * 2. User input for conditions to inspect.
+ * 3. Lists the result and is saved in a text file at /tmp/inspect/
+ */
+public class DurableLogInspectCommand extends DurableDataLogRepairCommand {
+
+    private static final String SEQUENCE_NUMBER = "SequenceNumber";
+    private static final String SEGMENT_ID = "SegmentId";
+    private static final String OFFSET = "Offset";
+    private static final String LENGTH = "Length";
+    private static final String ATTRIBUTES = "Attributes";
+    private static final String LESS_THAN = "<";
+    private static final String GREATER_THAN = ">";
+    private static final String LESS_THAN_EQUAL_TO = "<=";
+    private static final String GREATER_THAN_EQUAL_TO = ">=";
+    private static final String NOT_EQUAL_TO = "!=";
+    private static final String OPERATOR_AND = "and";
+    private static final String NEW_LINE_SEPARATOR = "\n";
+
+    /**
+     * Creates a new instance of the DurableLogInspectCommand class.
+     *
+     * @param args The arguments for the command.
+     */
+    public DurableLogInspectCommand(CommandArgs args) {
+        super(args);
+    }
+
+    @Override
+    public void execute() throws Exception {
+        ensureArgCount(1);
+        int containerId = getIntArg(0);
+        val bkConfig = getCommandArgs().getState().getConfigBuilder()
+                .include(BookKeeperConfig.builder().with(BookKeeperConfig.ZK_ADDRESS, getServiceConfig().getZkURL()))
+                .build().getConfig(BookKeeperConfig::builder);
+        @Cleanup
+        val zkClient = createZKClient();
+        @Cleanup
+        DurableDataLogFactory dataLogFactory = new BookKeeperLogFactory(bkConfig, zkClient, getCommandArgs().getState().getExecutor());
+        dataLogFactory.initialize();
+
+        // Open the Original Log in read-only mode.
+        @Cleanup
+        val originalDataLog = dataLogFactory.createDebugLogWrapper(containerId);
+
+        // Print the operations for the selected durableLog
+        int durableLogReadOperations = readDurableDataLog(containerId, originalDataLog);
+
+        output("Total reads original:" + durableLogReadOperations);
+        output("\n Note: Previous Inspect Result files if present will be deleted \n");
+
+        FileHelpers.deleteFileOrDirectory(new File(getCLIControllerConfig().getInspectResultLocation()));
+        //Get user input
```

Comment:
Comments go after a space and end in `.`.

Reply:
updated the comment.